app/models/journaled/outbox/event.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+module Journaled
+  module Outbox
+    # ActiveRecord model for database-backed event processing
+    #
+    # This model is only used when the Outbox::Adapter is configured.
+    # Events are stored in the database and processed by worker daemons.
+    #
+    # Successfully delivered events are deleted immediately.
+    class Event < Base
+      self.table_name = 'journaled_events'
+
+      scope :ready_to_process, -> {
+        where('locked_at IS NULL OR locked_at < ?', Time.current - Journaled.worker_lock_timeout.seconds)
+          .order(:created_at)
+      }
+
+      # Claim a batch of events for processing
+      #
+      # Uses SELECT FOR UPDATE SKIP LOCKED for distributed locking across multiple workers
+      #
+      # @param worker_id [String] Unique identifier for this worker (e.g., "hostname-pid")
+      # @return [Array<Journaled::Outbox::Event>] Claimed events
+      def self.claim_batch!(worker_id:)
+        if connection.adapter_name == 'PostgreSQL'
+          claim_batch_postgresql!(worker_id:)
+        else
+          claim_batch_generic!(worker_id:)
+        end
+      end
+
+      # Optimized PostgreSQL implementation using CTE with UPDATE ... FROM
+      def self.claim_batch_postgresql!(worker_id:)
+        timeout = Time.current - Journaled.worker_lock_timeout.seconds
+        batch_size = Journaled.worker_batch_size
+        locked_at = Time.current
+
+        sql = <<-SQL.squish
+          WITH locked_events AS (
+            SELECT id
+            FROM journaled_events
+            WHERE locked_at IS NULL OR locked_at < ?
+            ORDER BY created_at
+            LIMIT ?
+            FOR UPDATE SKIP LOCKED
+          )
+          UPDATE journaled_events
+          SET locked_at = ?,
+              locked_by = ?,
+              attempts = attempts + 1
+          FROM locked_events
+          WHERE journaled_events.id = locked_events.id
+          RETURNING journaled_events.*
+        SQL
+
+        find_by_sql([sql, timeout, batch_size, locked_at, worker_id])
+      end
+
+      # Generic implementation for other databases (SQLite, etc.)
+      def self.claim_batch_generic!(worker_id:)
+        transaction do
+          events = ready_to_process
+            .limit(Journaled.worker_batch_size)
+            .lock('FOR UPDATE SKIP LOCKED')
```

Comment:
I don't think this is standard sql (and I don't think sqlite supports it). Is this what delayed does?